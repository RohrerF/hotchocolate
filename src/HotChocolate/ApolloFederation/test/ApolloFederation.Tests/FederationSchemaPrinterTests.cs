using HotChocolate.Types;
using Snapshooter.Xunit;
using Xunit;

namespace HotChocolate.ApolloFederation
{
    public class FederationSchemaPrinterTests
    {
        [Fact]
        public void TestFederationPrinterApolloDirectivesSchemaFirst()
        {
            // arrange
            // todo

            // act
        }

        [Fact]
        public void TestFederationPrinterApolloDirectivesPureCodeFirst()
        {
            // arrange
            ISchema schema = SchemaBuilder.New()
                .AddApolloFederation()
                .AddQueryType<Query>()
                .Create();

            // check
            FederationSchemaPrinter.Print(schema).MatchSnapshot();
        }

        public class Query
        {
            public User GetEntity(int id) => default;
        }

        public class User
        {
            [Key]
            public int Id { get; set; }
            [External]
            public string IdCode { get; set; }
            [Requires("idCode")]
            public string IdCodeShort { get; set; }
            [Provides("zipcode")]
            public Address Address { get; set; }
        }

        public class Address
        {
            [External]
            public string Zipcode { get; set; }
        }
    }
}
