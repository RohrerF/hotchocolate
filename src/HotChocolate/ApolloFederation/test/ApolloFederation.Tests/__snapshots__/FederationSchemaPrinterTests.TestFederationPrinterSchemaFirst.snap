interface iQuery {
  someField(a: Int = null): TestType
}

type Query implements iQuery {
  someField(a: Int = null): TestType
}

type TestType @key(fields: "id") {
  id: Int!
  name: String!
  enum: SomeEnum
}

type TestTypeTwo {
  id: Int!
}

union TestTypes = TestType | TestTypeTwo

enum SomeEnum {
  FOO
  BAR
}
